grammar.js

```diff
@@ -963,6 +965,44 @@ const rules = {
       rep($.enumerator),
       '}',
     ),
+  /**
+   * Enum classes require their own rule, as they contain somewhat different
+   * syntax from enums. In particular cannot contain the `as` keyword.
+   */
+  enum_class_declaration: $ =>
+    seq(
+      opt($.attribute_modifier),
+      'enum',
+      'class',
+      field('name', $.identifier),
+      ':',
+      field('type', $._type),
+      opt($.extends_clause),
+      '{',
+      rep(seq($._type, $.identifier, '=', $._expression, ';')),
+      '}',
+    ),
+  /**
+   * Abstract enum classes have their own oddities. In particular they
+   * can contain abstract members.
+   */
+  abstract_enum_class_declaration: $ =>
+    seq(
+      opt($.attribute_modifier),
+      'abstract',
+      'enum',
+      'class',
+      field('name', $.identifier),
+      ':',
+      opt($.extends_clause),
+      field('type', $._type),
+      '{',
+      rep(choice(
+        seq($._type, $.identifier, '=', $._expression, ';'),
```

Comment:
Where possible and makes sense, we like to use smaller rules to build larger rules. For example, we could use a variation of `seq($._type, $.enumerator)` here instead (may want to change/rename nodes as desired).